diff -urN kde-baseapps-4.8.2.orig/dolphin/src/dolphinpart.desktop kde-baseapps-4.8.2/dolphin/src/dolphinpart.desktop
--- kde-baseapps-4.8.2.orig/dolphin/src/dolphinpart.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/dolphin/src/dolphinpart.desktop	2012-04-13 00:46:44.918778940 +0400
@@ -222,6 +222,7 @@
 Name[pl]=Kompaktowo
 Name[pt]=Compacto
 Name[pt_BR]=Compacto
+Name[ru]=Компактный
 Name[sk]=Kompaktný
 Name[sl]=Strnjeno
 Name[sr]=Сажето
diff -urN kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphingeneral.desktop kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphingeneral.desktop
--- kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphingeneral.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphingeneral.desktop	2012-04-12 22:05:15.482782002 +0400
@@ -272,7 +272,7 @@
 Comment[pt]=Configurar as opções gerais do gestor de ficheiros
 Comment[pt_BR]=Configurar opções gerais do gerenciador de arquivos
 Comment[ro]=Configurează opțiunile generale ale gestionarul de fișiere
-Comment[ru]=Настройка диспетчера файлов
+Comment[ru]=Основные настройки диспетчера файлов
 Comment[se]=Heivet oppalaš fiilagieđahallanheivehusat
 Comment[si]=සාමාන්‍ය ගොනු කළමාණාකරන සැකසුම් සකසන්න
 Comment[sk]=Všeobecné nastavenie správcu súborov
@@ -319,6 +319,7 @@
 X-KDE-Keywords[pl]=menedżer plików
 X-KDE-Keywords[pt]=gestor de ficheiros
 X-KDE-Keywords[pt_BR]=gerenciador de arquivos
+X-KDE-Keywords[ru]=file manager,диспетчер файлов,файловый менеджер,менеджер файлов
 X-KDE-Keywords[sk]=správca súborov
 X-KDE-Keywords[sl]=upravljalnik datotek
 X-KDE-Keywords[sr]=file manager,менаџер фајлова
diff -urN kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphinnavigation.desktop kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphinnavigation.desktop
--- kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphinnavigation.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphinnavigation.desktop	2012-04-12 22:05:15.485782001 +0400
@@ -319,6 +319,7 @@
 X-KDE-Keywords[pl]=menedżer plików
 X-KDE-Keywords[pt]=gestor de ficheiros
 X-KDE-Keywords[pt_BR]=gerenciador de arquivos
+X-KDE-Keywords[ru]=file manager,файловый менеджер,менеджер файлов,диспетчер файлов
 X-KDE-Keywords[sk]=správca súborov
 X-KDE-Keywords[sl]=upravljalnik datotek
 X-KDE-Keywords[sr]=file manager,менаџер фајлова
diff -urN kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphinservices.desktop kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphinservices.desktop
--- kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphinservices.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphinservices.desktop	2012-04-12 22:05:15.490782001 +0400
@@ -50,7 +50,7 @@
 Name[pt]=Serviços do Dolphin
 Name[pt_BR]=Serviços do Dolphin
 Name[ro]=Servicii Dolphin
-Name[ru]=Действия Dolphin
+Name[ru]=Службы Dolphin
 Name[si]=Dolphin සේවා
 Name[sk]=Služby Dolphinu
 Name[sl]=Dolphin - storitve
@@ -147,7 +147,7 @@
 Name[pt]=Serviços
 Name[pt_BR]=Serviços
 Name[ro]=Servicii
-Name[ru]=Действия
+Name[ru]=Службы
 Name[se]=Bálvalusat
 Name[si]=සේවා
 Name[sk]=Služby
@@ -224,7 +224,7 @@
 Comment[pt]=Configurar os serviços do gestor de ficheiros
 Comment[pt_BR]=Configurar serviços do gerenciador de arquivos
 Comment[ro]=Configurează serviciile gestionarului de fișiere
-Comment[ru]=Настройка действий в диспетчере файлов
+Comment[ru]=Настройка служб диспетчера файлов
 Comment[si]=ගොනු කළමණාකරුගේ සේවා සකසන්න
 Comment[sk]=Nastavenie služieb správcu súborov
 Comment[sl]=Nastavitve storitev upravljalnika datotek
@@ -270,6 +270,7 @@
 X-KDE-Keywords[pl]=menedżer plików
 X-KDE-Keywords[pt]=gestor de ficheiros
 X-KDE-Keywords[pt_BR]=gerenciador de arquivos
+X-KDE-Keywords[ru]=file manager,файловый менеджер,менеджер файлов,диспетчер файлов
 X-KDE-Keywords[sk]=správca súborov
 X-KDE-Keywords[sl]=upravljalnik datotek
 X-KDE-Keywords[sr]=file manager,менаџер фајлова
diff -urN kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphinviewmodes.desktop kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphinviewmodes.desktop
--- kde-baseapps-4.8.2.orig/dolphin/src/settings/kcm/kcmdolphinviewmodes.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/dolphin/src/settings/kcm/kcmdolphinviewmodes.desktop	2012-04-12 22:05:15.487781999 +0400
@@ -317,6 +317,7 @@
 X-KDE-Keywords[pl]=menedżer plików
 X-KDE-Keywords[pt]=gestor de ficheiros
 X-KDE-Keywords[pt_BR]=gerenciador de arquivos
+X-KDE-Keywords[ru]=file manager,файловый менеджер,менеджер файлов,диспетчер файлов
 X-KDE-Keywords[sk]=správca súborov
 X-KDE-Keywords[sl]=upravljalnik datotek
 X-KDE-Keywords[sr]=file manager,менаџер фајлова
diff -urN kde-baseapps-4.8.2.orig/kdepasswd/kcm/kcm_useraccount.desktop kde-baseapps-4.8.2/kdepasswd/kcm/kcm_useraccount.desktop
--- kde-baseapps-4.8.2.orig/kdepasswd/kcm/kcm_useraccount.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/kdepasswd/kcm/kcm_useraccount.desktop	2012-04-12 22:05:15.495782000 +0400
@@ -126,6 +126,7 @@
 X-KDE-Keywords[pl]=hasło;email;nazwa;organizacja;rzeczywista nazwa;obraz loginu;twarz; tryb echo;
 X-KDE-Keywords[pt]=senha;e-mail;nome;organização;nome verdadeiro;imagem de autenticação;cara;modo de eco;
 X-KDE-Keywords[pt_BR]=senha;e-mail;nome;organização;nome verdadeiro;imagem de autenticação;face;modo de eco;
+X-KDE-Keywords[ru]=password;email;name;organization;realname;login image;face; echo mode;пароль;электронная почта;имя;организация;настоящее имя;картинка;лицо;режим отображения
 X-KDE-Keywords[sk]=heslo;e-mail;meno;organizácia;skutočné meno;obrázok;tvár;režim výpisu;
 X-KDE-Keywords[sl]=geslo;pošta;ime;organizacija;pravo ime;slika ob prijavi;obraz;način odmeva;
 X-KDE-Keywords[sr]=password;email;name;organization;realname;login image;face; echo mode;лозинка;е‑пошта;име;организација;пуно име;пријавна слика;лице;ехо
diff -urN kde-baseapps-4.8.2.orig/konq-plugins/minitools/minitoolsplugin.desktop kde-baseapps-4.8.2/konq-plugins/minitools/minitoolsplugin.desktop
--- kde-baseapps-4.8.2.orig/konq-plugins/minitools/minitoolsplugin.desktop	2012-01-18 22:07:34.000000000 +0400
+++ kde-baseapps-4.8.2/konq-plugins/minitools/minitoolsplugin.desktop	2012-04-12 22:05:15.378782000 +0400
@@ -50,7 +50,7 @@
 Name[pt]=Mini-favoritos
 Name[pt_BR]=Minifavoritos
 Name[ro]=Bookmarklet-uri
-Name[ru]=Закладки
+Name[ru]=Букмарклеты
 Name[sk]=Bookmarklets (favelet)
 Name[sl]=Skripti v zaznamkih
 Name[sr]=Обележивачићи
@@ -106,7 +106,7 @@
 Comment[pt]=Activa o uso de mini-favoritos
 Comment[pt_BR]=Habilita o uso de minifavoritos
 Comment[ro]=Activează folosirea de bookmarklet-uri
-Comment[ru]=Включает использование закладок
+Comment[ru]=Включает использование букмарклетов
 Comment[sk]=Umožňuje použitie bookmarklets (favelet)
 Comment[sl]=Omogoči uporabo skriptov v zaznamkih
 Comment[sr]=Укључује обележивачиће
diff -urN kde-baseapps-4.8.2.orig/konq-plugins/uachanger/uachangerplugin.desktop kde-baseapps-4.8.2/konq-plugins/uachanger/uachangerplugin.desktop
--- kde-baseapps-4.8.2.orig/konq-plugins/uachanger/uachangerplugin.desktop	2011-09-26 13:43:17.000000000 +0400
+++ kde-baseapps-4.8.2/konq-plugins/uachanger/uachangerplugin.desktop	2012-04-12 22:05:15.283782001 +0400
@@ -51,7 +51,7 @@
 Name[pt]=Mudança de Agente do Utilizador
 Name[pt_BR]=Modificador do agente de usuário
 Name[ro]=Schimbă UserAgent
-Name[ru]=Строка UserAgent
+Name[ru]=Изменение UserAgent
 Name[sk]=Editor užívateľského agenta
 Name[sl]=Uporabniški posredniki
 Name[sr]=Измењивач корисничког агента
@@ -109,7 +109,7 @@
 Comment[pt]=Permite modificar o Agente de Utilizador detectado
 Comment[pt_BR]=Permite alterar o agente de usuário detectado
 Comment[ro]=Permite schimbare UserAgent-ului detectat
-Comment[ru]=Позволяет менять значение строки UserAgent
+Comment[ru]=Позволяет менять обнаруженного UserAgent
 Comment[sk]=Umožní upravovanie vybranému užívateľskému agentovi
 Comment[sl]=Omogoča spremembo zaznanega uporabniškega posrednika
 Comment[sr]=Омогућава измену одаслатог корисничког агента
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/ebrowsing/ebrowsing.desktop kde-baseapps-4.8.2/konqueror/settings/ebrowsing/ebrowsing.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/ebrowsing/ebrowsing.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/ebrowsing/ebrowsing.desktop	2012-04-12 22:05:15.258782002 +0400
@@ -77,7 +77,7 @@
 Name[pt]=Atalhos Web
 Name[pt_BR]=Atalhos da Internet
 Name[ro]=Acceleratori de web
-Name[ru]=Сокращения
+Name[ru]=Веб-сокращения
 Name[se]=Fierpmádatlávkestagat
 Name[si]=වෙබ් කෙටිමං
 Name[sk]=Webové skratky
@@ -169,7 +169,7 @@
 Comment[pt]=Configuração da navegação melhorada
 Comment[pt_BR]=Configura a navegação melhorada
 Comment[ro]=Configurează navigarea de web avansată
-Comment[ru]=Сокращения, используемые в адресной строке
+Comment[ru]=Настройка расширенного просмотра
 Comment[se]=Heivet viiddiduvvon fierbmegolgama
 Comment[si]=දියුණු කළ ගවේෂණ සැකසුම්
 Comment[sk]=Nastavenie rozšíreného prehliadania
@@ -219,6 +219,7 @@
 X-KDE-Keywords[pl]=Ulepszone przeglądanie,Przeglądanie,WWW,Internet,Filtry internetowe,Sieć,Wyszukiwarki,Silniki wyszukiwania,Skróty
 X-KDE-Keywords[pt]=Navegação Melhorada,Navegação,WWW,Internet,Filtros da Internet,Rede,Motores de Busca,Atalhos
 X-KDE-Keywords[pt_BR]=Navegação melhorada,Navegação,WWW,Internet,Filtros da Internet,Rede,Mecanismos de busca,Atalhos
+X-KDE-Keywords[ru]=Enhanced Browsing,Browsing,WWW,Internet,Internet Filters,Network,Search Engines,Shortcuts,расширенный просмотр,просмотр,интернет,интернет-фильтры,сеть,поисковые системы,сокращения
 X-KDE-Keywords[sk]=Rozšírené surfovanie,Surfovanie,WWW,Internet,Internetové filtre,Sieť,Vyhľadávacie nástroje,Odkazy
 X-KDE-Keywords[sr]=Enhanced Browsing,Browsing,WWW,Internet,Internet Filters,Network,Search Engines,Shortcuts,побољшано прегледање,прегледање,веб,интернет,филтери,мрежа,претраживачи,пречице
 X-KDE-Keywords[sr@ijekavian]=Enhanced Browsing,Browsing,WWW,Internet,Internet Filters,Network,Search Engines,Shortcuts,побољшано прегледање,прегледање,веб,интернет,филтери,мрежа,претраживачи,пречице
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/kio/bookmarks.desktop kde-baseapps-4.8.2/konqueror/settings/kio/bookmarks.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/kio/bookmarks.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/kio/bookmarks.desktop	2012-04-12 22:34:31.769781697 +0400
@@ -208,6 +208,7 @@
 X-KDE-Keywords[pl]=Zakładki
 X-KDE-Keywords[pt]=Favoritos
 X-KDE-Keywords[pt_BR]=Favoritos
+X-KDE-Keywords[ru]=Bookmarks,закладки,закладка
 X-KDE-Keywords[sk]=Záložky
 X-KDE-Keywords[sl]=Zaznamki
 X-KDE-Keywords[sr]=Bookmarks,обележивачи
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/kio/cache.desktop kde-baseapps-4.8.2/konqueror/settings/kio/cache.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/kio/cache.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/kio/cache.desktop	2012-04-12 22:05:15.144782000 +0400
@@ -170,7 +170,7 @@
 Comment[pt]=Configuração das opções da 'cache' da Web
 Comment[pt_BR]=Configura as opções de cache da Internet
 Comment[ro]=Configurează setările cache-ului web
-Comment[ru]=Настройка кэша
+Comment[ru]=Настройка параметров веб-кэша
 Comment[se]=Heivet fierpmádatgaskarádjusa
 Comment[si]=ජාල කැච් සැකසුම් සකසන්න
 Comment[sk]=Nastavenie vyrovnávacej pamäte pre WWW
@@ -221,6 +221,7 @@
 X-KDE-Keywords[pl]=Pamięć podręczna,Historia,Historia przeglądania,Porty,Rozmiar
 X-KDE-Keywords[pt]=Cache,Histórico,Histórico de Navegação,Portos,Tamanho
 X-KDE-Keywords[pt_BR]=Cache,Histórico,Histórico de navegação,Portas,Tamanho
+X-KDE-Keywords[ru]=Cache,History,Browsing History,Ports,Size,кэш,кеш,история,история браузера,история броузера,порты,размер
 X-KDE-Keywords[sk]=Cache,História,História surfovania,Porty,Veľkosť
 X-KDE-Keywords[sr]=Cache,History,Browsing History,Ports,Size,кеш,историјат,прегледање,портови,величина
 X-KDE-Keywords[sr@ijekavian]=Cache,History,Browsing History,Ports,Size,кеш,историјат,прегледање,портови,величина
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/kio/cookies.desktop kde-baseapps-4.8.2/konqueror/settings/kio/cookies.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/kio/cookies.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/kio/cookies.desktop	2012-04-12 22:05:15.139782001 +0400
@@ -217,6 +217,7 @@
 X-KDE-Keywords[pl]=Ciasteczka,Przeglądanie,Internet,WWW,Sieć,Polityka,Domeny
 X-KDE-Keywords[pt]=Cookies,Navegação,Internet,WWW,Rede,Política,Domínios
 X-KDE-Keywords[pt_BR]=Cookies,Navegação,Internet,WWW,Rede,Política,Domínios
+X-KDE-Keywords[ru]=Cookies,Browsing,Internet,WWW,Network,Policy,Domains,куки,просмотр,интернет,сеть,политика
 X-KDE-Keywords[sk]=Cookies,Sufrovanie,Internet,WWW,Sieť,Politika,Domény
 X-KDE-Keywords[sr]=Cookies,Browsing,Internet,WWW,Network,Policy,Domains,колачићи,прегледање,Интернет,веб,мрежа,смерница,домени
 X-KDE-Keywords[sr@ijekavian]=Cookies,Browsing,Internet,WWW,Network,Policy,Domains,колачићи,прегледање,Интернет,веб,мрежа,смерница,домени
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/kio/netpref.desktop kde-baseapps-4.8.2/konqueror/settings/kio/netpref.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/kio/netpref.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/kio/netpref.desktop	2012-04-12 22:05:15.227781999 +0400
@@ -78,7 +78,7 @@
 Name[pt]=Preferências da Ligação
 Name[pt_BR]=Preferências da conexão
 Name[ro]=Preferințe conexiune
-Name[ru]=Параметры сети
+Name[ru]=Параметры соединения
 Name[se]=Oktavuohta oidimat
 Name[si]=සම්බන්ධතා අභිමතයන්
 Name[sk]=Predvoľby pripojenia
@@ -218,6 +218,7 @@
 X-KDE-Keywords[pl]=czas oczekiwania,iopref,netpref,ustawienia sieci,ftp
 X-KDE-Keywords[pt]=tempo-limite,iopref,netpref,preferências de rede,ftp
 X-KDE-Keywords[pt_BR]=tempo-limite,iopref,netpref,preferências de rede,ftp
+X-KDE-Keywords[ru]=timeout,iopref,netpref,network preferences,ftp,истечение времени ожидания,таймаут,параметры сети
 X-KDE-Keywords[sk]=timeout,iopref,netpref,sieťové nastavenia,ftp
 X-KDE-Keywords[sr]=timeout,iopref,netpref,network preferences,ftp,прековреме,мрежне поставке,ФТП
 X-KDE-Keywords[sr@ijekavian]=timeout,iopref,netpref,network preferences,ftp,прековреме,мрежне поставке,ФТП
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/kio/proxy.desktop kde-baseapps-4.8.2/konqueror/settings/kio/proxy.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/kio/proxy.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/kio/proxy.desktop	2012-04-12 22:05:15.141782002 +0400
@@ -173,7 +173,7 @@
 Comment[pt]=Configurar os servidores 'proxy' usados
 Comment[pt_BR]=Configura os servidores proxy usados
 Comment[ro]=Configurează serverele proxy utilizate
-Comment[ru]=Настройка соединения через прокси-сервер
+Comment[ru]=Настройка прокси-сервера
 Comment[se]=Heivet gaskabálváid mat geavahuvvojit
 Comment[si]=භාවිතවන ප්‍රොක්සි සේවා දැකසුම්
 Comment[sk]=Nastavenie proxy serverov
@@ -222,6 +222,7 @@
 X-KDE-Keywords[pl]=Pośrednik,Serwer pośrednika,Zapora sieciowa,Squid,pośrednik
 X-KDE-Keywords[pt]=Proxy,servidor proxy,Firewall,Squid,proxy
 X-KDE-Keywords[pt_BR]=Proxy,Servidor proxy,Firewall,Squid,proxy
+X-KDE-Keywords[ru]=Proxy,Proxy server,Firewall,Squid,proxy,прокси,прокси-сервер,файервол,брандмауэр
 X-KDE-Keywords[sk]=Proxy,Proxy server,Firewall,Squid,proxy
 X-KDE-Keywords[sr]=Proxy,Proxy server,Firewall,Squid,proxy,прокси,Сквид,заштитни зид
 X-KDE-Keywords[sr@ijekavian]=Proxy,Proxy server,Firewall,Squid,proxy,прокси,Сквид,заштитни зид
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/kio/uasprovider.desktop kde-baseapps-4.8.2/konqueror/settings/kio/uasprovider.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/kio/uasprovider.desktop	2011-09-26 13:43:17.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/kio/uasprovider.desktop	2012-04-12 22:05:15.134782001 +0400
@@ -67,7 +67,7 @@
 Comment[pt]=Mensagens do Agente do Utilizador
 Comment[pt_BR]=Strings do agente de usuário
 Comment[ro]=Texte „Agent Utilizator”
-Comment[ru]=Версия браузера
+Comment[ru]=Строки UserAgent
 Comment[se]=Geavaheaddjeageanta-dovdamearkkat
 Comment[si]=පරිශීලක නියෝජිත වැකිය
 Comment[sk]=Reťazce WWW agenta
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/kio/useragent.desktop kde-baseapps-4.8.2/konqueror/settings/kio/useragent.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/kio/useragent.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/kio/useragent.desktop	2012-04-12 22:05:15.230782002 +0400
@@ -76,7 +76,7 @@
 Name[pt]=Identificação do Navegador
 Name[pt_BR]=Identificação do navegador
 Name[ro]=Identificare navigator web
-Name[ru]=Версия браузера
+Name[ru]=Идентификация браузера
 Name[se]=Fierpmádatloganidentifikašuvdna
 Name[si]=ගවේශකය හඳුනා ගැනීම
 Name[sk]=Identifikácia prehliadača
@@ -166,7 +166,7 @@
 Comment[pt]=Configuração da forma como o Konqueror se apresenta
 Comment[pt_BR]=Configura a forma como o Konqueror se apresenta
 Comment[ro]=Configurează modul în care Konqueror se identifică la serverele de web
-Comment[ru]=Настройка версии Konqueror
+Comment[ru]=Настройка идентификации Konqueror
 Comment[se]=Vállje mo Konqueror galgá iežas ovdanbuktit
 Comment[si]=Konqueror එය විසින්ම වාර්‍ථා කරන ආකාරය සකසන්න
 Comment[sk]=Nastavenie identifikácie Konquerora
@@ -214,6 +214,7 @@
 X-KDE-Keywords[pl]=Agent użytkownika,Przeglądarka,Internet,WWW,Sieć,agent, powiązania,serwer,login
 X-KDE-Keywords[pt]=Agente do Utilizador,Navegador,Internet,WWW,Rede,agente,interfaces,servidor,conta
 X-KDE-Keywords[pt_BR]=Agente do usuário,Navegador,Internet,WWW,Rede,agente,interfaces,servidor,conta
+X-KDE-Keywords[ru]=User Agent,Browser,Internet,WWW,Network,agent,bindings,server,login,пользовательский агент,браузер,интернет,сеть,агент,привязки,сервер,логин
 X-KDE-Keywords[sk]=Používateľský agent,Prehliadač,Internet,WWW,Sieť,agent,bindings,server,login
 X-KDE-Keywords[sr]=User Agent,Browser,Internet,WWW,Network,agent,bindings,server,login,кориснички агент,прегледач,Интернет,веб,мрежа,агент,свезе,сервер,пријава
 X-KDE-Keywords[sr@ijekavian]=User Agent,Browser,Internet,WWW,Network,agent,bindings,server,login,кориснички агент,прегледач,Интернет,веб,мрежа,агент,свезе,сервер,пријава
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konq/fileappearance.desktop kde-baseapps-4.8.2/konqueror/settings/konq/fileappearance.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konq/fileappearance.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konq/fileappearance.desktop	2012-04-12 22:05:15.237782000 +0400
@@ -219,6 +219,7 @@
 X-KDE-Keywords[pl]=konqueror,menadżer plików,zawijanie słów,zawijanie,podkreślenie,wyświetlanie,rozmiar pliku, bajty,czcionka,kolor,barwa
 X-KDE-Keywords[pt]=konqueror,gestor de ficheiros,mudança de linha,mudança,sublinhado,mostrar,tamanho do ficheiro,bytes,tipo de letra,cor
 X-KDE-Keywords[pt_BR]=konqueror,gerenciador de arquivos,mudança de linha,mudança,sublinhado,mostrar,tamanho do arquivo,bytes,fonte,cor,cores
+X-KDE-Keywords[ru]=konqueror,filemanager,word wrap,wrap,underline,display,filesize,bytes,font,color,colour,диспетчер файлов,файловый менеджер,менеджер файлов,перенос слов,перенос,подчеркивание,дисплей,размер файла,байт,шрифт,цвет
 X-KDE-Keywords[sk]=konqueror,správca súborov,zalamovanie slov,zalamovanie,podčiarknutie,displej,veľkosť súboru,bajty,písmo,farba,farby
 X-KDE-Keywords[sr]=konqueror,filemanager,word wrap,wrap,underline,display,filesize,bytes,font,color,colour,К‑освајач,менаџер фајлова,прелом речи,прелом,подвучено,приказ,величина фајла,бајтови,фонт,боја
 X-KDE-Keywords[sr@ijekavian]=konqueror,filemanager,word wrap,wrap,underline,display,filesize,bytes,font,color,colour,К‑освајач,менаџер фајлова,прелом речи,прелом,подвучено,приказ,величина фајла,бајтови,фонт,боја
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konq/filebehavior.desktop kde-baseapps-4.8.2/konqueror/settings/konq/filebehavior.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konq/filebehavior.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konq/filebehavior.desktop	2012-04-12 22:05:15.240782002 +0400
@@ -219,6 +219,7 @@
 X-KDE-Keywords[pl]=konqueror,menadżer plików,kfm,tips,podpowiedzi plików, osobne okno,wywołane okna,zachowanie,wykorzystanie pamięci, Domowy adres URL,operacje sieciowe,okno dialogowe postępu,kosz,śmietnik,usuń,potwierdzenie
 X-KDE-Keywords[pt]=konqueror,gestor de ficheiros,kfm,dicas,dicas do ficheiro,janela separada,janelas recriadas,comportamento,utilização de memória,URL inicial,operações de rede,janela de evolução,caixote do lixo,remoção,confirmação
 X-KDE-Keywords[pt_BR]=konqueror,gerenciador de arquivos,kfm,dicas,dicas do arquivo,janela separada,janelas recriadas,comportamento,utilização de memória,URL inicial,operações de rede,janela de evolução,lixeira,exclusão,confirmação
+X-KDE-Keywords[ru]=konqueror,filemanager,kfm,tips,file tips,separate window,spawn windows,behavior,behaviour,memory usage,Home URL,network operations,progress dialog,trash can,delete,confirmation,файловый менеджер,менеджер файлов,диспетчер файлов,ярлыки,ярлыки файлов,отдельное окно,порожденное окно,поведение,использование памяти,домашний адрес,сетевые операции,окно прогресса,корзина,удалить,подтверждение
 X-KDE-Keywords[sk]=konqueror,správca súborov,kfm,tipy,súborové tipy,oddelené okno,spawn okná,správanie,chovanie,využitie pamäte,Domovská URL,sieťové operácie,dialóg postupu,kôš,vymazať,potvrdenie
 X-KDE-Keywords[sr]=konqueror,filemanager,kfm,tips,file tips,separate window,spawn windows,behavior,behaviour,memory usage,Home URL,network operations,progress dialog,trash can,delete,confirmation,К‑освајач,менаџер фајлова,КФМ,савети,облачићи,засебни прозор,понашање,меморија,домаћи УРЛ,мрежни поступци,дијалог напретка,смеће,брисање,потврда
 X-KDE-Keywords[sr@ijekavian]=konqueror,filemanager,kfm,tips,file tips,separate window,spawn windows,behavior,behaviour,memory usage,Home URL,network operations,progress dialog,trash can,delete,confirmation,К‑освајач,менаџер фајлова,КФМ,савети,облачићи,засебни прозор,понашање,меморија,домаћи УРЛ,мрежни поступци,дијалог напретка,смеће,брисање,потврда
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konq/webappearance.desktop kde-baseapps-4.8.2/konqueror/settings/konq/webappearance.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konq/webappearance.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konq/webappearance.desktop	2012-04-12 22:05:15.235781999 +0400
@@ -220,6 +220,7 @@
 X-KDE-Keywords[pl]=konqueror,menadżer plików,zawijanie słów,zawijanie,podkreślenie,wyświetlanie,rozmiar pliku, bajty,czcionka,kolor,barwa
 X-KDE-Keywords[pt]=konqueror,gestor de ficheiros,mudança de linha,mudança,sublinhado,mostrar,tamanho do ficheiro,bytes,tipo de letra,cor
 X-KDE-Keywords[pt_BR]=konqueror,gerenciador de arquivos,mudança de linha,mudança,sublinhado,mostrar,tamanho do arquivo,bytes,fonte,cor,cores
+X-KDE-Keywords[ru]=konqueror,filemanager,word wrap,wrap,underline,display,filesize,bytes,font,color,colour,файловый менеджер,перенос текста,переход на новую строку,подчеркивание,дисплей,размер файла,байт,цвет
 X-KDE-Keywords[sk]=konqueror,správca súborov,zalamovanie slov,zalamovanie,podčiarknutie,displej,veľkosť súboru,bajty,písmo,farba,farby
 X-KDE-Keywords[sr]=konqueror,filemanager,word wrap,wrap,underline,display,filesize,bytes,font,color,colour,К‑освајач,менаџер фајлова,прелом речи,прелом,подвучено,приказ,величина фајла,бајтови,фонт,боја
 X-KDE-Keywords[sr@ijekavian]=konqueror,filemanager,word wrap,wrap,underline,display,filesize,bytes,font,color,colour,К‑освајач,менаџер фајлова,прелом речи,прелом,подвучено,приказ,величина фајла,бајтови,фонт,боја
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konq/webbehavior.desktop kde-baseapps-4.8.2/konqueror/settings/konq/webbehavior.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konq/webbehavior.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konq/webbehavior.desktop	2012-04-12 22:05:15.232782001 +0400
@@ -220,6 +220,7 @@
 X-KDE-Keywords[pl]=konqueror,menadżer plików,kfm,podpowiedzi,podpowiedzi plików, osobne okna,wywołane okna,zachowanie,wykorzystanie pamięci,Domowy adres URL,
 X-KDE-Keywords[pt]=konqueror,gestor de ficheiros,kfm,dicas,dicas de ficheiros,janela separada,janelas recriadas,comportamento,utilização de memória,URL inicial,
 X-KDE-Keywords[pt_BR]=konqueror,gerenciador de arquivos,kfm,dicas,dicas de arquivos,janela separada,janelas recriadas,comportamento,utilização de memória,URL inicial,
+X-KDE-Keywords[ru]=konqueror,filemanager,kfm,tips,file tips,separate window,spawn windows,behavior,behaviour,memory usage,Home URL,файловый менеджер,менеджер файлов,ярлыки,ярлыки файлов,отдельное окно,порожденные окна,поведение,использование памяти,домашний адрес
 X-KDE-Keywords[sk]=konqueror,správca súborov,kfm,tipy,súborové tipy,oddelené okno,spawn okná,správanie,chovanie,využitie pamäte,Domovská URL,
 X-KDE-Keywords[sr]=konqueror,filemanager,kfm,tips,file tips,separate window,spawn windows,behavior,behaviour,memory usage,Home URL,К‑освајач,менаџер фајлова,КФМ,савети,облачићи,засебни прозор,понашање,меморија,домаћи УРЛ
 X-KDE-Keywords[sr@ijekavian]=konqueror,filemanager,kfm,tips,file tips,separate window,spawn windows,behavior,behaviour,memory usage,Home URL,К‑освајач,менаџер фајлова,КФМ,савети,облачићи,засебни прозор,понашање,меморија,домаћи УРЛ
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/css/kcmcss.desktop kde-baseapps-4.8.2/konqueror/settings/konqhtml/css/kcmcss.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/css/kcmcss.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konqhtml/css/kcmcss.desktop	2012-04-12 22:05:15.243782000 +0400
@@ -76,7 +76,7 @@
 Name[pt]=Folhas de estilo
 Name[pt_BR]=Folhas de estilo
 Name[ro]=Foi de stil
-Name[ru]=Стили CSS
+Name[ru]=Таблицы стилей
 Name[se]=Stiilaárkkat
 Name[si]=හැඩපත
 Name[sk]=Štýly
@@ -168,7 +168,7 @@
 Comment[pt]=Configuração das folhas de estilo, usadas para mostrar as páginas Web
 Comment[pt_BR]=Configura as folhas de estilo utilizadas para renderizar as páginas da Internet
 Comment[ro]=Configurează foile de stil utilizate la randarea paginilor de web
-Comment[ru]=Настройка стилей CSS, используемых при показа веб-страниц
+Comment[ru]=Настройка таблицы стилей, используемых при показе веб-страниц
 Comment[se]=Heivet stiilaárkkaid mainna čájeha HTML-siidduid
 Comment[si]=වෙබ් පිටු පෙන්වීම සඳහා භාවිත කරන රටාපත්‍ර සකසන්න
 Comment[sk]=Nastavenie štýlov používaných pre zobrazovanie HTML
@@ -218,6 +218,7 @@
 X-KDE-Keywords[pl]=CSS,HTML,styl,arkusz stylów,dostępność
 X-KDE-Keywords[pt]=CSS,HTML,estilo,stylesheet,folha de estilo,acessibilidade
 X-KDE-Keywords[pt_BR]=CSS,HTML,estilo,planilha,folha de estilo,acessibilidade
+X-KDE-Keywords[ru]=CSS,HTML,style,stylesheet,accessibility,стиль, таблица стилей,доступность
 X-KDE-Keywords[sk]=CSS,HTML,štýl,stylesheet,prístupnosť
 X-KDE-Keywords[sr]=CSS,HTML,style,stylesheet,accessibility,ЦСС,ХТМЛ,стил,опис стила,приступачност
 X-KDE-Keywords[sr@ijekavian]=CSS,HTML,style,stylesheet,accessibility,ЦСС,ХТМЛ,стил,опис стила,приступачност
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_appearance.desktop kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_appearance.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_appearance.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_appearance.desktop	2012-04-12 22:05:15.251782001 +0400
@@ -156,7 +156,7 @@
 Comment[pt]=Configurar a apresentação das páginas Web
 Comment[pt_BR]=Configura a apresentação das páginas da Internet
 Comment[ro]=Configurează cum să fie afișate paginile web
-Comment[ru]=Настройка шрифтов веб-страниц
+Comment[ru]=Настройка отображения веб-страниц
 Comment[se]=Heivet man láhkai fierpmádatsiiddut čájehuvvot
 Comment[si]=වෙබ් පිටු පෙන්වන ආකාරය සකසන්න
 Comment[sk]=Nastavenie zobrazenia webových stránok
@@ -199,6 +199,7 @@
 X-KDE-Keywords[pl]=konqueror,kfm,przeglądarka,html,sieć,www,czcionki,zestawy znaków,kodowanie
 X-KDE-Keywords[pt]=konqueror,kfm,navegador,HTML,Web,WWW,tipos de letra,codificações,codificações de caracteres,codificação
 X-KDE-Keywords[pt_BR]=konqueror,kfm,navegador,html,web,www,fontes,codificações,codificações de caracteres,codificação
+X-KDE-Keywords[ru]=konqueror,kfm,browser,html,web,www,fonts,charsets,character sets,encoding,браузер,шрифты,кодировки,наборы символов,шифрование
 X-KDE-Keywords[sk]=konqueror,kfm,prehliadač,html,web,www,písma,znakové sady,sady znakov,kódovanie
 X-KDE-Keywords[sr]=konqueror,kfm,browser,html,web,www,fonts,charsets,character sets,encoding,К‑освајач,КФМ,прегледач,ХТМЛ,веб,фонт,кодирање,знак
 X-KDE-Keywords[sr@ijekavian]=konqueror,kfm,browser,html,web,www,fonts,charsets,character sets,encoding,К‑освајач,КФМ,прегледач,ХТМЛ,веб,фонт,кодирање,знак
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_behavior.desktop kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_behavior.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_behavior.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_behavior.desktop	2012-04-12 22:05:15.245781999 +0400
@@ -76,7 +76,7 @@
 Name[pt]=Comportamento na Web
 Name[pt_BR]=Comportamento da Web
 Name[ro]=Comportament web
-Name[ru]=Просмотр
+Name[ru]=Поведение интернет-браузера
 Name[se]=Webláhtten
 Name[si]=වෙබ් හැසිරීම
 Name[sk]=Správanie WWW
@@ -167,7 +167,7 @@
 Comment[pt]=Configurar o comportamento do navegador
 Comment[pt_BR]=Configura o comportamento do navegador
 Comment[ro]=Configurează comportamentul de navigator de web
-Comment[ru]=Настройка просмотра веб-страниц
+Comment[ru]=Настройка поведения браузера
 Comment[se]=Heivet mo fierpmádatlogan láhtte
 Comment[si]=වෙබ් හැසිරීම සැකසීම
 Comment[sk]=Nastavenie správania prehliadača
@@ -216,6 +216,7 @@
 X-KDE-Keywords[pl]=konqueror,kfm,przeglądarka,html,sieć,www,czcionki,kolory,barwy,java,javascript,kursor,odnośniki,obrazy,zestawy znaków,kodowanie
 X-KDE-Keywords[pt]=konqueror,kfm,navegador,HTML,Web,WWW,cores,Java,Javascript,cursor,ligações,imagens,codificações,codificações de caracteres,codificação
 X-KDE-Keywords[pt_BR]=konqueror,kfm,navegador,html,web,www,fontes,cores,java,javascript,cursor,links,imagens,codificações,codificações de caracteres,codificação
+X-KDE-Keywords[ru]=konqueror,kfm,browser,html,web,www,fonts,colours,colors,java,javascript,cursor,links,images,charsets,character sets,encoding,браузер,веб,шрифты,цвета,курсор,ссылки,изображения,кодировки,набор символов,шифрование
 X-KDE-Keywords[sk]=konqueror,kfm,prehliadač,html,web,www,písma,farba,farby,java,javascript,kurzor,odkazy,obrázky,znakové sady,sady znakov,kódovanie
 X-KDE-Keywords[sr]=konqueror,kfm,browser,html,web,www,fonts,colours,colors,java,javascript,cursor,links,images,charsets,character sets,encoding,К‑освајач,КФМ,прегледач,ХТМЛ,веб,фонт,боје,јава,јаваскрипт,показивач,курсор,везе,слике,кодирање
 X-KDE-Keywords[sr@ijekavian]=konqueror,kfm,browser,html,web,www,fonts,colours,colors,java,javascript,cursor,links,images,charsets,character sets,encoding,К‑освајач,КФМ,прегледач,ХТМЛ,веб,фонт,боје,јава,јаваскрипт,показивач,курсор,везе,слике,кодирање
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_general.desktop kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_general.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_general.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_general.desktop	2012-04-12 22:05:15.248782002 +0400
@@ -186,6 +186,7 @@
 X-KDE-Keywords[pl]=konqueror,karty
 X-KDE-Keywords[pt]=konqueror,páginas
 X-KDE-Keywords[pt_BR]=konqueror,abas
+X-KDE-Keywords[ru]=konqueror,tabs,вкладка,вкладки
 X-KDE-Keywords[sk]=konqueror,karty
 X-KDE-Keywords[sr]=konqueror,tabs,К‑освајач,језичци
 X-KDE-Keywords[sr@ijekavian]=konqueror,tabs,К‑освајач,језичци
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_java_js.desktop kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_java_js.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/konqhtml/khtml_java_js.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/konqhtml/khtml_java_js.desktop	2012-04-12 22:05:15.253781999 +0400
@@ -218,6 +218,7 @@
 X-KDE-Keywords[pl]=konqueror,kfm,przeglądarka,html,sieć,www,java,javascript
 X-KDE-Keywords[pt]=konqueror,kfm,navegador,HTML,Web,WWW,Java,Javascript
 X-KDE-Keywords[pt_BR]=konqueror,kfm,navegador,html,web,www,java,javascript
+X-KDE-Keywords[ru]=konqueror,kfm,browser,html,web,www,java,javascript,браузер,веб
 X-KDE-Keywords[sk]=konqueror,kfm,prehliadač,html,web,www,java,javascript
 X-KDE-Keywords[sr]=konqueror,kfm,browser,html,web,www,java,javascript,К‑освајач,КФМ,прегледач,ХТМЛ,веб,јава,јаваскрипт
 X-KDE-Keywords[sr@ijekavian]=konqueror,kfm,browser,html,web,www,java,javascript,К‑освајач,КФМ,прегледач,ХТМЛ,веб,јава,јаваскрипт
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/performance/kcmkonqyperformance.desktop kde-baseapps-4.8.2/konqueror/settings/performance/kcmkonqyperformance.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/performance/kcmkonqyperformance.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/performance/kcmkonqyperformance.desktop	2012-04-12 22:05:15.129782000 +0400
@@ -216,6 +216,7 @@
 X-KDE-Keywords[pl]=konqueror,ponowne wykorzystanie,wstępne wczytywanie
 X-KDE-Keywords[pt]=konqueror,reutilizar,pré-carregamento
 X-KDE-Keywords[pt_BR]=konqueror,reutilizar,pré-carregamento
+X-KDE-Keywords[ru]=konqueror,reuse,preloading,повторное использование,предварительная загрузка
 X-KDE-Keywords[sk]=konqueror,reuse,prednačítanie
 X-KDE-Keywords[sr]=konqueror,reuse,preloading,К‑освајач,предучитавање
 X-KDE-Keywords[sr@ijekavian]=konqueror,reuse,preloading,К‑освајач,предучитавање
diff -urN kde-baseapps-4.8.2.orig/konqueror/settings/performance/kcmperformance.desktop kde-baseapps-4.8.2/konqueror/settings/performance/kcmperformance.desktop
--- kde-baseapps-4.8.2.orig/konqueror/settings/performance/kcmperformance.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/settings/performance/kcmperformance.desktop	2012-04-12 22:05:15.126782000 +0400
@@ -214,6 +214,7 @@
 X-KDE-Keywords[pl]=wydajność,szybkość,prędkość,pamięć,ulepszenie
 X-KDE-Keywords[pt]=performance,velocidade,memória,melhoria
 X-KDE-Keywords[pt_BR]=desempenho,velocidade,memória,melhoria
+X-KDE-Keywords[ru]=performance,speed,memory,improve,быстродействие,производительность,скорость,память,повысить
 X-KDE-Keywords[sk]=výkon,rýchlosť,pamäť,zlepšenie
 X-KDE-Keywords[sr]=performance,speed,memory,improve,перформансе,брзина,меморија,побољшати
 X-KDE-Keywords[sr@ijekavian]=performance,speed,memory,improve,перформансе,брзина,меморија,побољшати
diff -urN kde-baseapps-4.8.2.orig/konqueror/sidebar/default_entries/places.desktop kde-baseapps-4.8.2/konqueror/sidebar/default_entries/places.desktop
--- kde-baseapps-4.8.2.orig/konqueror/sidebar/default_entries/places.desktop	2012-01-18 22:07:34.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/sidebar/default_entries/places.desktop	2012-04-12 22:05:15.104782002 +0400
@@ -122,7 +122,7 @@
 Comment[pt]=Esta é a lista de locais.
 Comment[pt_BR]=Esta é a lista de locais.
 Comment[ro]=Aceasta este lista de locuri.
-Comment[ru]=Панель «Точки входа»
+Comment[ru]=Список точек входа
 Comment[si]=මෙය ස්ථානවල ලැයිස්තුවයි.
 Comment[sk]=Toto je zoznam miest.
 Comment[sl]=To je seznam mest.
diff -urN kde-baseapps-4.8.2.orig/konqueror/sidebar/default_entries/services.desktop kde-baseapps-4.8.2/konqueror/sidebar/default_entries/services.desktop
--- kde-baseapps-4.8.2.orig/konqueror/sidebar/default_entries/services.desktop	2011-07-27 22:36:38.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/sidebar/default_entries/services.desktop	2012-04-12 22:05:15.110781999 +0400
@@ -69,7 +69,7 @@
 Name[pt]=Serviços
 Name[pt_BR]=Serviços
 Name[ro]=Servicii
-Name[ru]=Действия
+Name[ru]=Службы
 Name[se]=Bálvalusat
 Name[si]=සේවා
 Name[sk]=Služby
diff -urN kde-baseapps-4.8.2.orig/konqueror/sidebar/history_module/kcmhistory.desktop kde-baseapps-4.8.2/konqueror/sidebar/history_module/kcmhistory.desktop
--- kde-baseapps-4.8.2.orig/konqueror/sidebar/history_module/kcmhistory.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/konqueror/sidebar/history_module/kcmhistory.desktop	2012-04-12 22:05:15.093781996 +0400
@@ -216,6 +216,7 @@
 X-KDE-Keywords[pl]=historia,wygaśnięcie
 X-KDE-Keywords[pt]=histórico,expiração
 X-KDE-Keywords[pt_BR]=histórico,expiração
+X-KDE-Keywords[ru]=history,expire,журнал,история,устаревание
 X-KDE-Keywords[sk]=história,expirácia
 X-KDE-Keywords[sr]=history,expire,историјат,истицање
 X-KDE-Keywords[sr@ijekavian]=history,expire,историјат,истицање
diff -urN kde-baseapps-4.8.2.orig/lib/konq/konqdndpopupmenuplugin.desktop kde-baseapps-4.8.2/lib/konq/konqdndpopupmenuplugin.desktop
--- kde-baseapps-4.8.2.orig/lib/konq/konqdndpopupmenuplugin.desktop	2011-09-26 13:43:17.000000000 +0400
+++ kde-baseapps-4.8.2/lib/konq/konqdndpopupmenuplugin.desktop	2012-04-12 22:05:15.388781999 +0400
@@ -51,7 +51,7 @@
 Comment[pt]='Plugin' de 'drag-and-drop' para o Menu de Contexto do Konqueror.
 Comment[pt_BR]=Plug-in arrastar e soltar para os menus de contexto do Konqueror.
 Comment[ro]=Modul trage-și-lasă pentru meniul popup Konqueror.
-Comment[ru]=Расширение контекстного меню Konqueror
+Comment[ru]=Расширение перетаскивания для контекстного меню Konqueror
 Comment[si]=Konqueror පොප් අප් මෙනුව සඳහා ඇද-දැමීම් ප්ලගිනය
 Comment[sk]=Modul pre kontextové menu Konquerora.
 Comment[sl]=Vstavek povleci in spusti za Konquerorjev pojavni meni.
diff -urN kde-baseapps-4.8.2.orig/lib/konq/Templates/linkZIP.desktop kde-baseapps-4.8.2/lib/konq/Templates/linkZIP.desktop
--- kde-baseapps-4.8.2.orig/lib/konq/Templates/linkZIP.desktop	2012-01-10 18:14:25.000000000 +0400
+++ kde-baseapps-4.8.2/lib/konq/Templates/linkZIP.desktop	2012-04-12 22:05:15.421782001 +0400
@@ -64,7 +64,7 @@
 Name[pt]=Dispositivo ZIP...
 Name[pt_BR]=Dispositivo ZIP...
 Name[ro]=Dispozitiv ZIP...
-Name[ru]=ZIP...
+Name[ru]=Устройство ZIP...
 Name[se]=ZIP-ovttadat …
 Name[si]=ZIP උපාංගය...
 Name[sk]=Zariadenie ZIP...
diff -urN kde-baseapps-4.8.2.orig/nsplugins/khtml_plugins.desktop kde-baseapps-4.8.2/nsplugins/khtml_plugins.desktop
--- kde-baseapps-4.8.2.orig/nsplugins/khtml_plugins.desktop	2012-03-30 00:25:30.000000000 +0400
+++ kde-baseapps-4.8.2/nsplugins/khtml_plugins.desktop	2012-04-12 22:05:15.264782000 +0400
@@ -220,6 +220,7 @@
 X-KDE-Keywords[pl]=konqueror,kfm,przeglądarka,html,sieć,www,czcionki,wtyczki
 X-KDE-Keywords[pt]=konqueror,kfm,navegador,HTML,Web,WWW,tipos de letra,plugins
 X-KDE-Keywords[pt_BR]=konqueror,kfm,navegador,html,web,www,fontes,plug-ins
+X-KDE-Keywords[ru]=konqueror,kfm,browser,html,web,www,fonts,plugins,браузер,веб,шрифты,модули,плагины
 X-KDE-Keywords[sk]=konqueror,kfm,prehliadač,html,web,www,písma,pluginy
 X-KDE-Keywords[sr]=konqueror,kfm,browser,html,web,www,fonts,plugins,К‑освајач,КФМ,прегледач,ХТМЛ,веб,фонтови,прикључци
 X-KDE-Keywords[sr@ijekavian]=konqueror,kfm,browser,html,web,www,fonts,plugins,К‑освајач,КФМ,прегледач,ХТМЛ,веб,фонтови,прикључци
