diff -X exclude -Nuar kde-baseapps-4.8.1/dolphin/src/dolphincontextmenu.cpp kde-baseapps-4.8.1_new/dolphin/src/dolphincontextmenu.cpp
--- kde-baseapps-4.8.1/dolphin/src/dolphincontextmenu.cpp	2012-04-03 17:12:28.175919243 +0400
+++ kde-baseapps-4.8.1_new/dolphin/src/dolphincontextmenu.cpp	2012-04-03 17:13:08.939100035 +0400
@@ -225,11 +225,8 @@
     }
 }
 
-#include <KDebug>
-
 void DolphinContextMenu::openItemContextMenu()
 {
-    kDebug() << m_mainWindow->actionCollection()->actions();
     Q_ASSERT(!m_fileInfo.isNull());
 
     QAction* openParentInNewWindowAction = 0;
@@ -261,11 +258,11 @@
             //m_popup->addAction(m_mainWindow->actionCollection()->action("open_in_new_tab"));
 
             // insert 'Add to Places' entry
-            if (!placeExists(m_fileInfo.url())) {
-                addToPlacesAction = m_popup->addAction(KIcon("bookmark-new"),
-                                                       i18nc("@action:inmenu Add selected folder to places",
-                                                             "Add to Places"));
-            }
+            //if (!placeExists(m_fileInfo.url())) {
+            //    addToPlacesAction = m_popup->addAction(KIcon("bookmark-new"),
+            //                                           i18nc("@action:inmenu Add selected folder to places",
+            //                                                 "Add to Places"));
+            //}
 
             m_popup->addSeparator();
         } else if (m_baseUrl.protocol().contains("search")) {
@@ -290,10 +287,15 @@
         }
     }
 
+    insertDefaultItemActions();
+
     KFileItemActions fileItemActions;
     fileItemActions.setItemListProperties(selectedItemsProperties());
     addServiceActions(fileItemActions);
 
+    QAction* symlinkAction = m_mainWindow->actionCollection()->action("create_symlink");
+    m_popup->addAction(symlinkAction);
+
     //m_popup->addSeparator();
 
     insertDefaultItemActions();
diff -X exclude -Nuar kde-baseapps-4.8.1/dolphin/src/views/dolphinviewactionhandler.cpp kde-baseapps-4.8.1_new/dolphin/src/views/dolphinviewactionhandler.cpp
--- kde-baseapps-4.8.1/dolphin/src/views/dolphinviewactionhandler.cpp	2012-04-03 17:12:28.177919497 +0400
+++ kde-baseapps-4.8.1_new/dolphin/src/views/dolphinviewactionhandler.cpp	2012-04-03 17:11:43.504241704 +0400
@@ -35,8 +35,10 @@
 #include <KToggleAction>
 #include <KRun>
 #include <KPropertiesDialog>
+#include <QDir>
 
 #include <KDebug>
+#include "unistd.h"
 
 DolphinViewActionHandler::DolphinViewActionHandler(KActionCollection* collection, QObject* parent)
     : QObject(parent),
@@ -111,6 +113,10 @@
     rename->setIcon(KIcon("edit-rename"));
     connect(rename, SIGNAL(triggered()), this, SLOT(slotRename()));
 
+    KAction* symlink = m_actionCollection->addAction("create_symlink");
+    symlink->setText(i18nc("@action:inmenu File", "Create symlink"));
+    connect(symlink, SIGNAL(triggered()), this, SLOT(slotSymlink()));
+
     KAction* moveToTrash = m_actionCollection->addAction("move_to_trash");
     moveToTrash->setText(i18nc("@action:inmenu File", "Move to Trash"));
     moveToTrash->setIcon(KIcon("user-trash"));
@@ -282,6 +288,27 @@
     m_currentView->renameSelectedItems();
 }
 
+void DolphinViewActionHandler::slotSymlink()
+{
+    QDir currentDir(m_currentView->selectedItems().at(0).url().path());
+    KUrl url;
+    KFileItem fi = m_currentView->selectedItems().at(0);
+
+//    if(m_currentView->selectedItems().at(0).isDir())
+    currentDir.cdUp();
+
+    url = currentDir.absolutePath();
+
+    for(int i = 0; i < m_currentView->selectedItems().urlList().count(); i++)
+    {
+        KUrl tUrl = m_currentView->selectedItems().urlList().at(i);
+        KUrl destUrl = tUrl;
+        destUrl.setFileName(i18n("Link to ") + destUrl.fileName());
+
+        symlink(tUrl.path().toLocal8Bit().data(), destUrl.path().toLocal8Bit().data());
+    }
+}
+
 void DolphinViewActionHandler::slotTrashActivated(Qt::MouseButtons, Qt::KeyboardModifiers modifiers)
 {
     emit actionBeingHandled();
diff -X exclude -Nuar kde-baseapps-4.8.1/dolphin/src/views/dolphinviewactionhandler.h kde-baseapps-4.8.1_new/dolphin/src/views/dolphinviewactionhandler.h
--- kde-baseapps-4.8.1/dolphin/src/views/dolphinviewactionhandler.h	2012-04-03 17:12:28.177919497 +0400
+++ kde-baseapps-4.8.1_new/dolphin/src/views/dolphinviewactionhandler.h	2012-04-03 17:11:43.504241704 +0400
@@ -124,6 +124,11 @@
     void slotDeleteItems();
 
     /**
+     * Creates symbolic links to selected files named "Link to ..."
+     */
+    void slotSymlink();
+
+    /**
      * Switches between showing a preview of the file content and showing the icon.
      */
     void togglePreview(bool);
