--- a/dolphin/src/panels/places/placesitemmodel.cpp
+++ b/dolphin/src/panels/places/placesitemmodel.cpp
@@ -650,6 +650,7 @@
                     found = true;
                     if (newBookmark.metaDataItem("UDI").isEmpty()) {
                         item->setBookmark(newBookmark);
+                        item->setText(i18nc("KFile System Bookmarks", newBookmark.text().toUtf8().data()));
                     }
                     break;
                 }

