diff -urN kde-baseapps-4.11.0/dolphin/src/CMakeLists.txt kde-baseapps-4.11.0-patched/dolphin/src/CMakeLists.txt
--- kde-baseapps-4.11.0/dolphin/src/CMakeLists.txt	2013-08-15 12:43:26.133008620 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/CMakeLists.txt	2013-08-15 12:44:00.303680294 +1100
@@ -48,6 +48,7 @@
     kitemviews/kitemlistgroupheader.cpp
     kitemviews/kitemlistheader.cpp
     kitemviews/kitemlistselectionmanager.cpp
+    kitemviews/kitemlistklooktoggle.cpp
     kitemviews/kitemliststyleoption.cpp
     kitemviews/kitemlistview.cpp
     kitemviews/kitemlistviewaccessible.cpp
diff -urN kde-baseapps-4.11.0/dolphin/src/dolphinmainwindow.cpp kde-baseapps-4.11.0-patched/dolphin/src/dolphinmainwindow.cpp
--- kde-baseapps-4.11.0/dolphin/src/dolphinmainwindow.cpp	2013-08-15 12:43:26.133008620 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/dolphinmainwindow.cpp	2013-08-15 12:44:00.304680336 +1100
@@ -85,6 +85,8 @@
 #include <QClipboard>
 #include <QToolButton>
 #include <QSplitter>
+#include <QProcess>
+#include <QApplication>
 
 namespace {
     // Used for GeneralSettings::version() to determine whether
@@ -118,7 +120,8 @@
     m_settingsDialog(),
     m_controlButton(0),
     m_updateToolBarTimer(0),
-    m_lastHandleUrlStatJob(0)
+    m_lastHandleUrlStatJob(0),
+    m_isFocusOnTerminal(false)
 {
     setObjectName("Dolphin#");
 
@@ -231,6 +234,8 @@
     if (!showMenu) {
         createControlButton();
     }
+
+    connect (qApp, SIGNAL(focusChanged(QWidget*,QWidget*)), this, SLOT(focusChanged(QWidget*,QWidget*)));
 }
 
 DolphinMainWindow::~DolphinMainWindow()
@@ -395,6 +400,15 @@
         compareFilesAction->setEnabled(false);
     }
 
+
+    //KLook
+    QAction* openInKLook = actionCollection()->action("open_in_KLook");
+    if ((selectedUrlsCount > 0) && (!m_isFocusOnTerminal)){
+        openInKLook->setEnabled(true);
+    } else {
+        openInKLook->setEnabled(false);
+    }
+
     emit selectionChanged(selection);
 }
 
@@ -1660,6 +1674,14 @@
     openInNewWindow->setText(i18nc("@action:inmenu", "Open in New Window"));
     openInNewWindow->setIcon(KIcon("window-new"));
     connect(openInNewWindow, SIGNAL(triggered()), this, SLOT(openInNewWindow()));
+
+    // KLook
+    KAction* openInKLook = actionCollection()->addAction("open_in_KLook");
+    openInKLook->setText(i18nc("@action:inmenu", "Preview"));
+    openInKLook->setShortcut(Qt::Key_Space);
+    openInKLook->setIcon(KIcon("klook"));
+    openInKLook->setEnabled(false);
+    connect(openInKLook, SIGNAL(triggered()), this, SLOT(previewItem()));
 }
 
 void DolphinMainWindow::setupDockWidgets()
@@ -1999,6 +2021,9 @@
             this, SLOT(goBack()));
     connect(view, SIGNAL(goForwardRequested()),
             this, SLOT(goForward()));
+    connect(view, SIGNAL(klookTrigger()),
+            this, SLOT(previewItem()));
+
 
     const KUrlNavigator* navigator = container->urlNavigator();
     connect(navigator, SIGNAL(urlChanged(KUrl)),
@@ -2149,4 +2174,51 @@
     }
 }
 
+void DolphinMainWindow::previewItem()
+{
+    KFileItemList list;
+    int index = 0;
+    if ( m_activeViewContainer->view()->selectedItemsCount() > 1) {
+        list = m_activeViewContainer->view()->selectedItems();
+    } else {
+        list = m_activeViewContainer->view()->sortedItems();
+        index = list.indexOf( m_activeViewContainer->view()->selectedItems().at(0));
+    }
+    QStringList urlList;
+    foreach(KFileItem item, list)
+    {
+        urlList.append(item.url().url());
+    }
+    urlList.append("-i " + QString::number(index , 10));
+    if (!list.isEmpty())
+    {
+        QProcess* process = new QProcess();
+        connect(process,SIGNAL(finished(int)),process,SLOT(deleteLater()));
+        process->start(QString("klook"), urlList);
+    }
+}
+
+void DolphinMainWindow::focusChanged(QWidget *old, QWidget *now)
+{
+#ifndef Q_OS_WIN
+    if (now)
+    {
+        if (QString(now->metaObject()->className()).contains("Konsole::TerminalDisplay"))
+        {
+            QAction* openInKLook = actionCollection()->action("open_in_KLook");
+            m_isFocusOnTerminal = true;
+            openInKLook->setEnabled(false);
+        }
+    }
+    if (old)
+    {
+        if (QString(old->metaObject()->className()).contains("Konsole::TerminalDisplay"))
+        {
+            QAction* openInKLook = actionCollection()->action("open_in_KLook");
+            m_isFocusOnTerminal = false;
+            openInKLook->setEnabled(true);
+        }
+    }
+#endif
+}
 #include "dolphinmainwindow.moc"
diff -urN kde-baseapps-4.11.0/dolphin/src/dolphinmainwindow.h kde-baseapps-4.11.0-patched/dolphin/src/dolphinmainwindow.h
--- kde-baseapps-4.11.0/dolphin/src/dolphinmainwindow.h	2013-08-15 12:43:26.129008541 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/dolphinmainwindow.h	2013-08-15 12:44:00.305680380 +1100
@@ -36,6 +36,7 @@
 
 typedef KIO::FileUndoManager::CommandType CommandType;
 
+class DolphinDockWidget;
 class DolphinViewActionHandler;
 class DolphinApplication;
 class DolphinSettingsDialog;
@@ -51,6 +52,7 @@
 class QSplitter;
 class QToolButton;
 class QVBoxLayout;
+class Panel;
 
 /**
  * @short Main window for Dolphin.
@@ -102,6 +104,8 @@
      */
     KNewFileMenu* newFileMenu() const;
 
+    void focusChanged(QWidget *old, QWidget *now);
+
 public slots:
     /**
      * Pastes the clipboard data into the currently selected folder
@@ -372,6 +376,12 @@
     void openInNewTab();
 
     /**
+     * Opens the selected files in KLook.
+     */
+
+    void previewItem();
+
+    /**
      * Opens the selected folder in a new window.
      */
     void openInNewWindow();
@@ -593,6 +603,7 @@
     QTimer* m_updateToolBarTimer;
 
     KIO::Job* m_lastHandleUrlStatJob;
+    bool m_isFocusOnTerminal;
 };
 
 inline DolphinViewContainer* DolphinMainWindow::activeViewContainer() const
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kfileitemlistwidget.cpp kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kfileitemlistwidget.cpp
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kfileitemlistwidget.cpp	2013-08-15 12:43:26.130008561 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kfileitemlistwidget.cpp	2013-08-15 12:44:00.305680380 +1100
@@ -70,6 +70,44 @@
     return text;
 }
 
+QRectF KFileItemListWidget::klookToggleRect() const
+{
+    const_cast<KFileItemListWidget *>(this)->triggerCacheRefreshing();
+
+    const int iconHeight = styleOption().iconSize;
+
+    int toggleSize = KIconLoader::SizeSmall;
+    if (iconHeight >= KIconLoader::SizeEnormous) {
+        toggleSize = KIconLoader::SizeMedium;
+    } else if (iconHeight >= KIconLoader::SizeLarge) {
+        toggleSize = KIconLoader::SizeSmallMedium;
+    }
+
+    QPointF pos = QPointF(iconRect().topRight().rx() - toggleSize, iconRect().topRight().ry());
+
+    // If the selection toggle has a very small distance to the
+    // widget borders, the size of the selection toggle will get
+    // increased to prevent an accidental clicking of the item
+    // when trying to hit the toggle.
+
+    const int widgetHeight = size().height();
+    const int widgetWidth = size().width();
+    const int minMargin = 2;
+
+    if (toggleSize + minMargin * 2 >= widgetHeight) {
+        pos.rx() -= (widgetHeight - toggleSize) / 2;
+        toggleSize = widgetHeight;
+        pos.setY(0);
+    }
+    if (toggleSize + minMargin * 2 >= widgetWidth) {
+        pos.ry() -= (widgetWidth - toggleSize) / 2;
+        toggleSize = widgetWidth;
+        pos.setX(0);
+    }
+
+    return QRectF(pos, QSizeF(toggleSize, toggleSize));
+}
+
 KFileItemListWidget::KFileItemListWidget(KItemListWidgetInformant* informant, QGraphicsItem* parent) :
     KStandardItemListWidget(informant, parent)
 {
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kfileitemlistwidget.h kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kfileitemlistwidget.h
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kfileitemlistwidget.h	2013-08-15 12:43:26.129008541 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kfileitemlistwidget.h	2013-08-15 12:44:00.305680380 +1100
@@ -43,6 +43,7 @@
     virtual ~KFileItemListWidget();
 
     static KItemListWidgetInformant* createInformant();
+    virtual QRectF klookToggleRect() const;
 
 protected:
     virtual bool isRoleRightAligned(const QByteArray& role) const;
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistcontroller.cpp kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistcontroller.cpp
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistcontroller.cpp	2013-08-15 12:43:26.130008561 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistcontroller.cpp	2013-08-15 12:44:00.306680423 +1100
@@ -544,6 +544,13 @@
         m_selectionManager->beginAnchoredSelection(m_pressedIndex);
         return true;
     }
+    if (m_view->isAboveKlookToggle(m_pressedIndex, m_pressedMousePos)){
+        m_selectionManager->setSelected(m_pressedIndex, 1, KItemListSelectionManager::Select);
+        m_selectionManager->setCurrentItem(m_pressedIndex);
+        m_selectionManager->beginAnchoredSelection(m_pressedIndex);
+        emit klookTrigger();
+        return true;
+    }
 
     const bool shiftPressed = event->modifiers() & Qt::ShiftModifier;
     const bool controlPressed = event->modifiers() & Qt::ControlModifier;
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistcontroller.h kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistcontroller.h
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistcontroller.h	2013-08-15 12:43:26.130008561 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistcontroller.h	2013-08-15 12:44:00.306680423 +1100
@@ -233,6 +233,8 @@
     void modelChanged(KItemModelBase* current, KItemModelBase* previous);
     void viewChanged(KItemListView* current, KItemListView* previous);
 
+    void klookTrigger();
+
 private slots:
     void slotViewScrollOffsetChanged(qreal current, qreal previous);
 
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistklooktoggle.cpp kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistklooktoggle.cpp
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistklooktoggle.cpp	1970-01-01 10:00:00.000000000 +1000
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistklooktoggle.cpp	2013-08-15 12:44:00.306680423 +1100
@@ -0,0 +1,106 @@
+/***************************************************************************
+ *   Copyright (C) 2011 by Peter Penz <peter.penz19@gmail.com>             *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA            *
+ ***************************************************************************/
+
+#include "kitemlistklooktoggle_p.h"
+
+#include <KIconEffect>
+#include <KIconLoader>
+#include <QPainter>
+
+#include <KDebug>
+
+KItemListKlookToggle::KItemListKlookToggle(QGraphicsItem* parent) :
+    QGraphicsWidget(parent, 0),    
+    m_hovered(false)
+{
+    setAcceptHoverEvents(true);
+}
+
+KItemListKlookToggle::~KItemListKlookToggle()
+{
+}
+/*
+void KItemListKlookToggle::setChecked(bool checked)
+{
+    if (m_checked != checked) {
+        m_checked = checked;
+        m_pixmap = QPixmap();
+        update();
+    }
+}
+*/
+/*
+bool KItemListKlookToggle::isChecked() const
+{
+    return m_checked;
+}
+*/
+void KItemListKlookToggle::paint(QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget)
+{
+    Q_UNUSED(option);
+    Q_UNUSED(widget);
+
+    if (m_pixmap.isNull()) {
+        updatePixmap();
+    }
+
+    const qreal x = (size().width()  - qreal(m_pixmap.width()))  / 2;
+    const qreal y = (size().height() - qreal(m_pixmap.height())) / 2;
+    painter->drawPixmap(x, y, m_pixmap);
+}
+
+void KItemListKlookToggle::hoverEnterEvent(QGraphicsSceneHoverEvent* event)
+{
+    QGraphicsWidget::hoverEnterEvent(event);
+    m_hovered = true;
+    m_pixmap = QPixmap();
+}
+
+void KItemListKlookToggle::hoverLeaveEvent(QGraphicsSceneHoverEvent* event)
+{
+    QGraphicsWidget::hoverLeaveEvent(event);
+    m_hovered = false;
+    m_pixmap = QPixmap();
+}
+
+void KItemListKlookToggle::updatePixmap()
+{
+    const char* icon = "klook";
+
+    int iconSize = qMin(size().width(), size().height());
+    if (iconSize < KIconLoader::SizeSmallMedium) {
+        iconSize = KIconLoader::SizeSmall;
+    } else if (iconSize < KIconLoader::SizeMedium) {
+        iconSize = KIconLoader::SizeSmallMedium;
+    } else if (iconSize < KIconLoader::SizeLarge) {
+        iconSize = KIconLoader::SizeMedium;
+    } else if (iconSize < KIconLoader::SizeHuge) {
+        iconSize = KIconLoader::SizeLarge;
+    } else if (iconSize < KIconLoader::SizeEnormous) {
+        iconSize = KIconLoader::SizeHuge;
+    }
+
+    m_pixmap = KIconLoader::global()->loadIcon(QLatin1String(icon), KIconLoader::NoGroup, iconSize);
+
+    if (m_hovered) {
+        KIconLoader::global()->iconEffect()->apply(m_pixmap, KIconLoader::Desktop, KIconLoader::ActiveState);
+    }
+}
+
+#include "kitemlistklooktoggle_p.moc"
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistklooktoggle_p.h kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistklooktoggle_p.h
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistklooktoggle_p.h	1970-01-01 10:00:00.000000000 +1000
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistklooktoggle_p.h	2013-08-15 12:44:00.306680423 +1100
@@ -0,0 +1,61 @@
+/***************************************************************************
+ *   Copyright (C) 2011 by Peter Penz <peter.penz19@gmail.com>             *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA            *
+ ***************************************************************************/
+
+#ifndef KITEMLISTKLOOKTOGGLE_H
+#define KITEMLISTKLOOKTOGGLE_H
+
+#include <libdolphin_export.h>
+
+#include <QGraphicsWidget>
+#include <QPixmap>
+
+class QPropertyAnimation;
+
+/**
+ * @brief Allows to launch klook by clicking on icon overlay
+ */
+class LIBDOLPHINPRIVATE_EXPORT KItemListKlookToggle : public QGraphicsWidget
+{
+    Q_OBJECT
+
+public:
+    KItemListKlookToggle(QGraphicsItem* parent);
+    virtual ~KItemListKlookToggle();
+
+    //void setChecked(bool checked);
+    //bool isChecked() const;
+
+    virtual void paint(QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget = 0);
+
+protected:
+    virtual void hoverEnterEvent(QGraphicsSceneHoverEvent* event);
+    virtual void hoverLeaveEvent(QGraphicsSceneHoverEvent* event);
+
+private:
+    void updatePixmap();
+
+private:
+    //bool m_checked;
+    bool m_hovered;
+    QPixmap m_pixmap;
+};
+
+#endif
+
+
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistview.cpp kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistview.cpp
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistview.cpp	2013-08-15 12:43:26.130008561 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistview.cpp	2013-08-15 12:44:00.308680501 +1100
@@ -75,6 +75,7 @@
 KItemListView::KItemListView(QGraphicsWidget* parent) :
     QGraphicsWidget(parent),
     m_enabledSelectionToggles(false),
+    m_enabledKlookToggles(false),
     m_grouped(false),
     m_supportsItemExpanding(false),
     m_editingRole(false),
@@ -299,6 +300,25 @@
     return m_enabledSelectionToggles;
 }
 
+void KItemListView::setEnabledKlookToggles(bool enabled)
+{
+    if (m_enabledKlookToggles != enabled) {
+        m_enabledKlookToggles = enabled;
+
+        QHashIterator<int, KItemListWidget*> it(m_visibleItems);
+        while (it.hasNext()) {
+            it.next();
+            it.value()->setEnabledKlookToggle(enabled);
+        }
+    }
+}
+
+bool KItemListView::enabledKlookToggles() const
+{
+    return m_enabledKlookToggles;
+}
+
+
 KItemListController* KItemListView::controller() const
 {
     return m_controller;
@@ -450,6 +470,25 @@
     return false;
 }
 
+bool KItemListView::isAboveKlookToggle(int index, const QPointF& pos) const
+{
+
+    if (!m_enabledKlookToggles) {
+        return false;
+    }
+
+    const KItemListWidget* widget = m_visibleItems.value(index);
+    if (widget) {
+        const QRectF selectionToggleRect = widget->klookToggleRect();
+        if (!selectionToggleRect.isEmpty()) {
+            const QPointF mappedPos = widget->mapFromItem(this, pos);
+            return selectionToggleRect.contains(mappedPos);
+        }
+    }
+    return false;
+}
+
+
 int KItemListView::firstVisibleIndex() const
 {
     return m_layouter->firstVisibleIndex();
@@ -1926,6 +1965,7 @@
     widget->setSelected(selectionManager->isSelected(index));
     widget->setHovered(false);
     widget->setEnabledSelectionToggle(enabledSelectionToggles());
+    widget->setEnabledKlookToggle(enabledKlookToggles());
     widget->setIndex(index);
     widget->setData(m_model->data(index));
     widget->setSiblingsInformation(QBitArray());
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistview.h kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistview.h
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistview.h	2013-08-15 12:43:26.131008581 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistview.h	2013-08-15 12:44:00.308680501 +1100
@@ -112,6 +112,9 @@
     void setEnabledSelectionToggles(bool enabled);
     bool enabledSelectionToggles() const;
 
+    void setEnabledKlookToggles(bool enabled);
+    bool enabledKlookToggles() const;
+
     /**
      * @return Controller of the item-list. The controller gets
      *         initialized by KItemListController::setView() and will
@@ -181,6 +184,7 @@
     bool isAboveSelectionToggle(int index, const QPointF& pos) const;
     bool isAboveExpansionToggle(int index, const QPointF& pos) const;
 
+    bool isAboveKlookToggle(int index, const QPointF& pos) const;
     /**
      * @return Index of the first item that is at least partly visible.
      *         -1 is returned if the model contains no items.
@@ -699,6 +703,7 @@
 
 private:
     bool m_enabledSelectionToggles;
+    bool m_enabledKlookToggles;
     bool m_grouped;
     bool m_supportsItemExpanding;
     bool m_editingRole;
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistwidget.cpp kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistwidget.cpp
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistwidget.cpp	2013-08-15 12:43:26.130008561 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistwidget.cpp	2013-08-15 12:44:00.309680536 +1100
@@ -22,6 +22,7 @@
 
 #include "kitemlistwidget.h"
 
+#include "kitemlistklooktoggle_p.h"
 #include "kitemlistview.h"
 #include "kitemmodelbase.h"
 
@@ -52,6 +53,7 @@
     m_hovered(false),
     m_alternateBackground(false),
     m_enabledSelectionToggle(false),
+    m_enabledKlookToggle(false),
     m_data(),
     m_visibleRoles(),
     m_columnWidths(),
@@ -61,7 +63,8 @@
     m_hoverCache(0),
     m_hoverAnimation(0),
     m_selectionToggle(0),
-    m_editedRole()
+    m_editedRole(),
+    m_klookToggle(0)
 {
 }
 
@@ -76,6 +79,9 @@
         delete m_selectionToggle;
         m_selectionToggle = 0;
 
+        delete m_klookToggle;
+        m_klookToggle = 0;
+
         if (m_hoverAnimation) {
             m_hoverAnimation->stop();
             m_hoverOpacity = 0;
@@ -263,6 +269,9 @@
         if (m_enabledSelectionToggle && !(QApplication::mouseButtons() & Qt::LeftButton)) {
             initializeSelectionToggle();
         }
+        if (m_enabledKlookToggle && !(QApplication::mouseButtons() & Qt::LeftButton)) {
+            initializeKlookToggle();
+        }
     } else {
         m_hoverAnimation->setStartValue(hoverOpacity());
         m_hoverAnimation->setEndValue(0.0);
@@ -306,6 +315,19 @@
     return m_enabledSelectionToggle;
 }
 
+void KItemListWidget::setEnabledKlookToggle(bool enable)
+{
+    if (m_enabledKlookToggle != enable) {
+        m_enabledKlookToggle = enable;
+        update();
+    }
+}
+
+bool KItemListWidget::enabledKlookToggle() const
+{
+    return m_enabledKlookToggle;
+}
+
 void KItemListWidget::setSiblingsInformation(const QBitArray& siblings)
 {
     const QBitArray previous = m_siblingsInfo;
@@ -355,6 +377,12 @@
     return QRectF();
 }
 
+QRectF KItemListWidget::klookToggleRect() const
+{
+    return QRectF();
+}
+
+
 QRectF KItemListWidget::expansionToggleRect() const
 {
     return QRectF();
@@ -463,6 +491,8 @@
     if (!m_hovered) {
         delete m_selectionToggle;
         m_selectionToggle = 0;
+        delete m_klookToggle;
+        m_klookToggle = 0;
     }
 }
 
@@ -481,12 +511,29 @@
     m_selectionToggle->setChecked(isSelected());
 }
 
+void KItemListWidget::initializeKlookToggle()
+{
+    Q_ASSERT(m_enabledKlookToggle);
+
+    if (!m_klookToggle) {
+        m_klookToggle = new KItemListKlookToggle(this);
+    }
+
+    const QRectF toggleRect = klookToggleRect();;
+    m_klookToggle->setPos(toggleRect.topLeft());
+    m_klookToggle->resize(toggleRect.size());
+
+}
+
 void KItemListWidget::setHoverOpacity(qreal opacity)
 {
     m_hoverOpacity = opacity;
     if (m_selectionToggle) {
         m_selectionToggle->setOpacity(opacity);
     }
+    if (m_klookToggle) {
+        m_klookToggle->setOpacity(opacity);
+    }
 
     if (m_hoverOpacity <= 0.0) {
         delete m_hoverCache;
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistwidget.h kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistwidget.h
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kitemlistwidget.h	2013-08-15 12:43:26.130008561 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kitemlistwidget.h	2013-08-15 12:44:00.309680536 +1100
@@ -33,6 +33,7 @@
 
 class KItemListSelectionToggle;
 class KItemListView;
+class KItemListKlookToggle;
 class QPropertyAnimation;
 
 /**
@@ -114,6 +115,9 @@
     void setEnabledSelectionToggle(bool enabled);
     bool enabledSelectionToggle() const;
 
+    void setEnabledKlookToggle(bool enable);
+    bool enabledKlookToggle() const;
+
     /**
      * Sets the sibling information for the item and all of its parents.
      * The sibling information of the upper most parent is represented by
@@ -181,6 +185,12 @@
      */
     virtual QPixmap createDragPixmap(const QStyleOptionGraphicsItem* option, QWidget* widget = 0);
 
+     /**
+     * @return Rectangle for the klook-toggle that is used for item preview.
+     */
+    virtual QRectF klookToggleRect() const;
+
+
 signals:
     void roleEditingCanceled(int index, const QByteArray& role, const QVariant& value);
     void roleEditingFinished(int index, const QByteArray& role, const QVariant& value);
@@ -211,6 +221,7 @@
 
 private:
     void initializeSelectionToggle();
+    void initializeKlookToggle();
     void setHoverOpacity(qreal opacity);
     void clearHoverCache();
     void drawItemStyleOption(QPainter* painter, QWidget* widget, QStyle::State styleState);
@@ -225,6 +236,7 @@
     bool m_hovered;
     bool m_alternateBackground;
     bool m_enabledSelectionToggle;
+    bool m_enabledKlookToggle;
     QHash<QByteArray, QVariant> m_data;
     QList<QByteArray> m_visibleRoles;
     QHash<QByteArray, qreal> m_columnWidths;
@@ -238,6 +250,8 @@
     KItemListSelectionToggle* m_selectionToggle;
 
     QByteArray m_editedRole;
+    
+    KItemListKlookToggle* m_klookToggle;
 };
 
 inline const KItemListWidgetInformant* KItemListWidget::informant() const
diff -urN kde-baseapps-4.11.0/dolphin/src/kitemviews/kstandarditemlistwidget.h kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kstandarditemlistwidget.h
--- kde-baseapps-4.11.0/dolphin/src/kitemviews/kstandarditemlistwidget.h	2013-08-15 12:43:26.132008601 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/kitemviews/kstandarditemlistwidget.h	2013-08-15 12:44:00.309680536 +1100
@@ -157,6 +157,7 @@
     virtual void resizeEvent(QGraphicsSceneResizeEvent* event);
     virtual void showEvent(QShowEvent* event);
     virtual void hideEvent(QHideEvent* event);
+    void triggerCacheRefreshing();
 
 private slots:
     void slotCutItemsChanged();
@@ -164,7 +165,6 @@
     void slotRoleEditingFinished(const QByteArray& role, const QVariant& value);
 
 private:
-    void triggerCacheRefreshing();
     void updateExpansionArea();
     void updatePixmapCache();
 
diff -urN kde-baseapps-4.11.0/dolphin/src/settings/dolphin_generalsettings.kcfg kde-baseapps-4.11.0-patched/dolphin/src/settings/dolphin_generalsettings.kcfg
--- kde-baseapps-4.11.0/dolphin/src/settings/dolphin_generalsettings.kcfg	2013-08-15 12:43:26.133008620 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/settings/dolphin_generalsettings.kcfg	2013-08-15 12:44:00.309680536 +1100
@@ -61,6 +61,10 @@
             <label>Show selection toggle</label>
             <default>true</default>
         </entry>
+        <entry name="ShowKlookToggle" type="Bool">
+            <label>Show preview toggle</label>
+            <default>false</default>
+        </entry>
         <entry name="ShowToolTips" type="Bool">
             <label>Show tooltips</label>
             <default>false</default>
diff -urN kde-baseapps-4.11.0/dolphin/src/settings/general/behaviorsettingspage.cpp kde-baseapps-4.11.0-patched/dolphin/src/settings/general/behaviorsettingspage.cpp
--- kde-baseapps-4.11.0/dolphin/src/settings/general/behaviorsettingspage.cpp	2013-08-15 12:43:26.134008640 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/settings/general/behaviorsettingspage.cpp	2013-08-15 12:44:00.310680568 +1100
@@ -43,7 +43,8 @@
     m_showToolTips(0),
     m_showSelectionToggle(0),
     m_naturalSorting(0),
-    m_renameInline(0)
+    m_renameInline(0),
+    m_showKlookToggle(0)
 {
     QVBoxLayout* topLayout = new QVBoxLayout(this);
 
@@ -64,6 +65,9 @@
     // 'Show selection marker'
     m_showSelectionToggle = new QCheckBox(i18nc("@option:check", "Show selection marker"), this);
 
+    // 'Show klook toggle'
+    m_showKlookToggle = new QCheckBox(i18nc("@option:check", "Show preview marker"), this);
+
     // 'Natural sorting of items'
     m_naturalSorting = new QCheckBox(i18nc("option:check", "Natural sorting of items"), this);
 
@@ -73,6 +77,7 @@
     topLayout->addWidget(viewPropsBox);
     topLayout->addWidget(m_showToolTips);
     topLayout->addWidget(m_showSelectionToggle);
+    topLayout->addWidget(m_showKlookToggle);
     topLayout->addWidget(m_naturalSorting);
     topLayout->addWidget(m_renameInline);
     topLayout->addStretch();
@@ -83,6 +88,7 @@
     connect(m_globalViewProps, SIGNAL(toggled(bool)), this, SIGNAL(changed()));
     connect(m_showToolTips, SIGNAL(toggled(bool)), this, SIGNAL(changed()));
     connect(m_showSelectionToggle, SIGNAL(toggled(bool)), this, SIGNAL(changed()));
+    connect(m_showKlookToggle, SIGNAL(toggled(bool)), this, SIGNAL(changed()));
     connect(m_naturalSorting, SIGNAL(toggled(bool)), this, SIGNAL(changed()));
     connect(m_renameInline, SIGNAL(toggled(bool)), this, SIGNAL(changed()));
 }
@@ -113,6 +119,7 @@
         globalProps.setDirProperties(props);
     }
 
+    settings->setShowKlookToggle(m_showKlookToggle->isChecked());
     const bool naturalSorting = m_naturalSorting->isChecked();
     if (KGlobalSettings::naturalSorting() != naturalSorting) {
         KConfigGroup group(KGlobal::config(), "KDE");
@@ -137,6 +144,7 @@
 
     m_showToolTips->setChecked(GeneralSettings::showToolTips());
     m_showSelectionToggle->setChecked(GeneralSettings::showSelectionToggle());
+    m_showKlookToggle->setChecked(GeneralSettings::showKlookToggle());
     m_naturalSorting->setChecked(KGlobalSettings::naturalSorting());
     m_renameInline->setChecked(GeneralSettings::renameInline());
 }
diff -urN kde-baseapps-4.11.0/dolphin/src/settings/general/behaviorsettingspage.h kde-baseapps-4.11.0-patched/dolphin/src/settings/general/behaviorsettingspage.h
--- kde-baseapps-4.11.0/dolphin/src/settings/general/behaviorsettingspage.h	2013-08-15 12:43:26.134008640 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/settings/general/behaviorsettingspage.h	2013-08-15 12:44:00.310680568 +1100
@@ -59,6 +59,7 @@
     QCheckBox* m_showSelectionToggle;
     QCheckBox* m_naturalSorting;
     QCheckBox* m_renameInline;
+    QCheckBox* m_showKlookToggle;
 };
 
 #endif
diff -urN kde-baseapps-4.11.0/dolphin/src/views/dolphinitemlistview.cpp kde-baseapps-4.11.0-patched/dolphin/src/views/dolphinitemlistview.cpp
--- kde-baseapps-4.11.0/dolphin/src/views/dolphinitemlistview.cpp	2013-08-15 12:43:26.128008522 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/views/dolphinitemlistview.cpp	2013-08-15 12:44:00.310680568 +1100
@@ -41,6 +41,7 @@
     KFileItemListView(parent),
     m_zoomLevel(0)
 {
+    setEnabledKlookToggles(GeneralSettings::showKlookToggle());
     updateFont();
     updateGridSize();
 }
@@ -89,6 +90,7 @@
     beginTransaction();
 
     setEnabledSelectionToggles(GeneralSettings::showSelectionToggle());
+    setEnabledKlookToggles(GeneralSettings::showKlookToggle());
 
     const bool expandableFolders = (itemLayout() == KFileItemListView::DetailsLayout) &&
                                    DetailsModeSettings::expandableFolders();
diff -urN kde-baseapps-4.11.0/dolphin/src/views/dolphinview.cpp kde-baseapps-4.11.0-patched/dolphin/src/views/dolphinview.cpp
--- kde-baseapps-4.11.0/dolphin/src/views/dolphinview.cpp	2013-08-15 12:43:26.128008522 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/views/dolphinview.cpp	2013-08-15 12:44:00.311680594 +1100
@@ -155,6 +155,7 @@
     connect(controller, SIGNAL(itemUnhovered(int)), this, SLOT(slotItemUnhovered(int)));
     connect(controller, SIGNAL(itemDropEvent(int,QGraphicsSceneDragDropEvent*)), this, SLOT(slotItemDropEvent(int,QGraphicsSceneDragDropEvent*)));
     connect(controller, SIGNAL(modelChanged(KItemModelBase*,KItemModelBase*)), this, SLOT(slotModelChanged(KItemModelBase*,KItemModelBase*)));
+    connect(controller, SIGNAL(klookTrigger()), this, SIGNAL(klookTrigger()));
 
     connect(m_model, SIGNAL(directoryLoadingStarted()),       this, SLOT(slotDirectoryLoadingStarted()));
     connect(m_model, SIGNAL(directoryLoadingCompleted()),     this, SLOT(slotDirectoryLoadingCompleted()));
@@ -1690,4 +1691,14 @@
     return url;
 }
 
+KFileItemList DolphinView::sortedItems() const
+{
+    KFileItemList  list;
+    for (int i = 0; i < m_model->count(); i++) {
+        list.append(m_model->fileItem(i));
+    }
+    return list;
+}
+
+
 #include "dolphinview.moc"
diff -urN kde-baseapps-4.11.0/dolphin/src/views/dolphinview.h kde-baseapps-4.11.0-patched/dolphin/src/views/dolphinview.h
--- kde-baseapps-4.11.0/dolphin/src/views/dolphinview.h	2013-08-15 12:43:26.128008522 +1100
+++ kde-baseapps-4.11.0-patched/dolphin/src/views/dolphinview.h	2013-08-15 12:44:00.311680594 +1100
@@ -303,6 +303,8 @@
      */
     void setViewPropertiesContext(const QString& context);
     QString viewPropertiesContext() const;
+    
+    KFileItemList sortedItems() const;
 
     /**
      * Checks if the given \a item can be opened as folder (e.g. archives).
@@ -534,6 +536,8 @@
      */
     void writeStateChanged(bool isFolderWritable);
 
+    void klookTrigger();
+
     /**
      * Is emitted if the URL should be changed to the previous URL of the
      * history (e.g. because the "back"-mousebutton has been pressed).
