--- kde-baseapps-4.9.98/dolphin/src/panels/places/placesitemmodel.cpp.fileplaces	2012-12-17 15:29:48.000000000 +0100
+++ kde-baseapps-4.9.98/dolphin/src/panels/places/placesitemmodel.cpp	2013-01-29 20:00:48.132765255 +0100
@@ -35,6 +35,7 @@
 #include <KLocale>
 #include <KStandardDirs>
 #include <KUser>
+#include <KGlobalSettings>
 #include "placesitem.h"
 #include <QAction>
 #include <QDate>
@@ -917,6 +918,23 @@ void PlacesItemModel::createSystemBookma
                                                 "user-trash",
                                                 I18N_NOOP2("KFile System Bookmarks", "Trash")));
 
+    // We add Documents, Music, Downloads, Pictures, Videos to be in sync with kdelibs
+    m_systemBookmarks.append(SystemBookmarkData(KUrl(KGlobalSettings::documentPath()),
+                                                "folder-documents",
+                                                I18N_NOOP2("KFile System Bookmarks", "Documents")));
+    m_systemBookmarks.append(SystemBookmarkData(KUrl(KGlobalSettings::musicPath()),
+                                                "folder-sound",
+                                                I18N_NOOP2("KFile System Bookmarks", "Music")));
+    m_systemBookmarks.append(SystemBookmarkData(KUrl(KGlobalSettings::downloadPath()),
+                                                "folder-downloads",
+                                                I18N_NOOP2("KFile System Bookmarks", "Downloads")));
+    m_systemBookmarks.append(SystemBookmarkData(KUrl(KGlobalSettings::videosPath()),
+                                                "folder-video",
+                                                I18N_NOOP2("KFile System Bookmarks", "Videos")));
+    m_systemBookmarks.append(SystemBookmarkData(KUrl(KGlobalSettings::picturesPath()),
+                                                "folder-image",
+                                                I18N_NOOP2("KFile System Bookmarks", "Pictures")));
+
     if (m_fileIndexingEnabled) {
         m_systemBookmarks.append(SystemBookmarkData(KUrl("timeline:/today"),
                                                     "go-jump-today",
