diff -urN kde-baseapps-4.8.1.orig/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg kde-baseapps-4.8.1/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg
--- kde-baseapps-4.8.1.orig/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg	2012-03-28 14:34:18.575110634 +0400
+++ kde-baseapps-4.8.1/dolphin/src/settings/dolphin_directoryviewpropertysettings.kcfg	2012-03-28 14:34:39.595507193 +0400
@@ -31,7 +31,7 @@
         <entry name="PreviewsShown" type="Bool" >
             <label context="@label">Previews shown</label>
             <whatsthis context="@info:whatsthis">When this option is enabled, a preview of the file content is shown as an icon.</whatsthis>
-            <default>false</default>
+            <default>true</default>
         </entry>
 
         <entry name="GroupedSorting" type="Bool" >
